routes/noteRoutes.js

```diff
@@ -0,0 +1,110 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const Note = require("../models/Note");
+const router = express.Router();
+
+
+router.get("/", async (req, res) => {
+    try {
+        const page = parseInt(req.query.page) || 1;  
```

Comment:
You should handle cases when the user requests a page number that exceeds the total pages.

```
if (page > Math.ceil(totalNotes / limit)) {
    return res.status(404).json({ success: false, message: "Page not found" });
}
```